chore: Sync markdown to DITA, volume 2 (#1255)
kyma-project/telemetry-manager: docs/user/03-traces.md

### Telemetry Manager

The TracePipeline resource is watched by Telemetry Manager, which is responsible for generating the custom parts of the OTel Collector configuration.

![Manager resources](./assets/traces-resources.drawio.svg)

- Telemetry Manager watches all TracePipeline resources and related Secrets.
- Whenever the configuration changes, it validates the configuration and generates a new configuration for OTel Collector, where a ConfigMap for the configuration is generated.
- Referenced Secrets are copied into one Secret that is mounted to the OTel Collector as well.
- Furthermore, Telemetry Manager takes care of the full lifecycle of the OTel Collector Deployment itself. Only if there is a TracePipeline defined, the collector is deployed.

If you don't want to use the Traces feature, simply don't set up a TracePipeline.

## Setting up a TracePipeline

In the following steps, you can see how to construct and deploy a typical TracePipeline. Learn more about the available [parameters and attributes](resources/04-tracepipeline.md).

> ### Step 1a. Create a TracePipeline With an OTLP GRPC Output

Author: agreed! Thank you for noticing!

### Telemetry Manager

The TracePipeline resource is watched by Telemetry Manager, which is responsible for generating the custom parts of the OTel Collector configuration.

![Manager resources](./assets/traces-resources.drawio.svg)

- Telemetry Manager watches all TracePipeline resources and related Secrets.
- Whenever the configuration changes, it validates the configuration and generates a new configuration for OTel Collector, where a ConfigMap for the configuration is generated.
- Referenced Secrets are copied into one Secret that is mounted to the OTel Collector as well.
- Furthermore, Telemetry Manager takes care of the full lifecycle of the OTel Collector Deployment itself. Only if there is a TracePipeline defined, the collector is deployed.

If you don't want to use the Traces feature, simply don't set up a TracePipeline.

## Setting up a TracePipeline

In the following steps, you can see how to construct and deploy a typical TracePipeline. Learn more about the available [parameters and attributes](resources/04-tracepipeline.md).

> ### 1a. Create a TracePipeline With an OTLP GRPC Output

Reviewer: ```suggestion
### 1. Create a TracePipeline With an OTLP GRPC Output
```

    randomSamplingPercentage: 5.00
```

### 3a: Add Authentication Details From Plain Text

To integrate with external systems, you must configure authentication  details. You can use mutual TLS (mTLS), Basic Authentication, or custom headers:

<!-- tabs:start -->

#### **Mutual TLS**

```yaml

> apiVersion: telemetry.kyma-project.io/v1

Reviewer 2: That is wrong, only the istio resource version changed!
```suggestion
apiVersion: telemetry.kyma-project.io/v1alpha1
```

Author: 🙈

#### **Basic Authentication**

```yaml

> apiVersion: telemetry.kyma-project.io/v1

Reviewer 2: ```suggestion
apiVersion: telemetry.kyma-project.io/v1alpha1
```

#### **Token-Based With Custom Headers**

```yaml

> apiVersion: telemetry.kyma-project.io/v1

Reviewer 2: ```suggestion
apiVersion: telemetry.kyma-project.io/v1alpha1
```

<!-- tabs:end -->

### 3b: Add Authentication Details From Secrets

Integrations into external systems usually need authentication details dealing with sensitive data. To handle that data properly in Secrets, TracePipeline supports the reference of Secrets.

Using the **valueFrom** attribute, you can map Secret keys for mutual TLS (mTLS), Basic Authentication, or with custom headers.

You can store the value of the token in the referenced Secret without any prefix or scheme, and you can configure it in the `headers` section of the TracePipeline. In the following example, the token has the prefix "Bearer".

<!-- tabs:start -->

#### **Mutual TLS**

```yaml

> apiVersion: telemetry.kyma-project.io/v1

Reviewer 2: ```suggestion
apiVersion: telemetry.kyma-project.io/v1alpha1
```

#### **Basic Authentication**

```yaml

> apiVersion: telemetry.kyma-project.io/v1

Reviewer 2: ```suggestion
apiVersion: telemetry.kyma-project.io/v1alpha1
```

#### **Token-Based With Custom Headers**

```yaml

> apiVersion: telemetry.kyma-project.io/v1

Reviewer 2: ```suggestion
apiVersion: telemetry.kyma-project.io/v1alpha1
```

backend   True                      True              True           2m
```

## Kyma Modules With Tracing Capabilities

Kyma bundles several modules that can be involved in user flows. Applications involved in a distributed trace must propagate the trace context to keep the trace complete. Optionally, they can enrich the trace with custom spans, which requires reporting them to the backend.

### Istio

The Istio module is crucial in distributed tracing because it provides the [ingress gateway](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/). Usually, this is where external requests enter the cluster scope and are enriched with trace context if it hasn't happened yet. Furthermore, every component that's part of the Istio Service Mesh runs an Istio proxy, which propagates the context properly but also creates span data. If Istio tracing is activated and taking care of trace propagation in your application, you get a complete picture of a trace, because every component automatically contributes span data. Also, Istio tracing is pre-configured to be based on the vendor-neutral [w3c-tracecontext](https://www.w3.org/TR/trace-context/) protocol.

> > [!WARNING]

Reviewer 2: Warning should be removed

  token: YYY
```

### Step 4: Rotate the Secret

Telemetry Manager continuously watches the Secret referenced with the **secretKeyRef** construct. You can update the Secret’s values, and Telemetry Manager detects the changes and applies the new Secret to the setup.

> [!TIP]
> If you use a Secret owned by the [SAP BTP Operator](https://github.com/SAP/sap-btp-service-operator), you can configure an automated rotation using a `credentialsRotationPolicy` with a specific `rotationFrequency` and don’t have to intervene manually.

### Step 5: Deploy the Pipeline

To activate the TracePipeline, apply the `tracepipeline.yaml`  resource file in your cluster:

```bash
kubectl apply -f tracepipeline.yaml
```

### Result

You activated a TracePipeline and traces start streaming to your backend.

> To check that the pipeline is running, verify that the status of the TracePipeline in your cluster have status `True`:

Author: thank you! I aligned it.

```bash
kubectl get tracepipeline
NAME      CONFIGURATION GENERATED   GATEWAY HEALTHY   FLOW HEALTHY   AGE
backend   True                      True              True           2m
```

## Kyma Modules With Tracing Capabilities

Kyma bundles several modules that can be involved in user flows. Applications involved in a distributed trace must propagate the trace context to keep the trace complete. Optionally, they can enrich the trace with custom spans, which requires reporting them to the backend.

### Istio

> The Istio module is crucial in distributed tracing because it provides the [ingress gateway](https://istio.io/latest/docs/tasks/traffic-management/ingress/ingress-control/). Typically, this is where external requests enter the cluster scope and are enriched with trace context if it hasn't happened earlier. Furthermore, every component that's part of the Istio Service Mesh runs an Istio proxy, which propagates the context properly but also creates span data.

Author: agreed!

### Eventing

The Kyma [Eventing](https://kyma-project.io/#/eventing-manager/user/README) module uses the [CloudEvents](https://cloudevents.io/) protocol (which natively supports the [W3C Trace Context](https://www.w3.org/TR/trace-context) propagation). Because of that, it propagates trace context properly. However, it doesn't enrich a trace with more advanced span data.

### Serverless

By default, all engines for the [Serverless](https://kyma-project.io/#/serverless-manager/user/README) module integrate the [Open Telemetry SDK](https://opentelemetry.io/docs/reference/specification/metrics/sdk/). Thus, the used middlewares are configured to automatically propagate the trace context for chained calls.

Because the Telemetry endpoints are configured by default, Serverless also reports custom spans for incoming and outgoing requests. You can [customize Function traces](https://kyma-project.io/#/serverless-manager/user/tutorials/01-100-customize-function-traces) to add more spans as part of your Serverless source code.

## Operations

> A TracePipeline creates a Deployment running OTel Collector instances in your cluster. That Deployment serves OTLP endpoints and ships received data to the configured backend.

Reviewer 2: ```suggestion
A TracePipeline manages several OTel Collector instances in your cluster. That Deployment serves OTLP endpoints and ships received data to the configured backend.
```

The Telemetry module ensures that the OTel Collector instances are operational and healthy at any time, for example, with buffering and retries. However, there may be situations when the instances drop traces, or cannot handle the trace load.

> To detect and fix such situations, check the pipeline status and check out [Troubleshooting](#troubleshooting).

Author: Thank you! I aligned it.
